osm2streets/src/transform/classify_intersections.rs

```diff
@@ -101,18 +100,15 @@ pub fn guess_complexity(
         .iter()
         .map(|(s, d)| (roads[*s].id, roads[*d].id))
         .collect();
-    match worst_conflict {
-        Cross => (Crossing, Cross, movements),
-        c => (
-            if roads.len() == 2 {
-                Connection
-            } else {
-                MultiConnection
-            },
-            c,
-            movements,
-        ),
-    }
+    (
+        match worst_conflict {
+            Uncontested => Connection,
+            Diverge => Fork,
+            Merge => Fork, // TODO check for give way signs or count lanes to detect Intersections.
```

Comment:
There are some cases that look like the "merge" version of a `Fork` when looking at the half-road/`Movement` level, but should be `Intersection`s when looking at the lane-level, or when considering yield signs. The classic example is a slip lane that merges into the destination road with a *Give Way* sign. [Here's an example](https://www.openstreetmap.org/node/1836742019) (without the give way sign tagged, as is common currently), and [just to the north](https://www.openstreetmap.org/node/31337970) is a sliplane that joins with a "lane added" sign, so should be a `Fork`.

![forks and intersections](https://user-images.githubusercontent.com/3784591/203306376-57ecb427-e812-48da-be42-8dfc27001f18.png)

Another example I've seen is a [T-intersection between two oneway roads](https://www.openstreetmap.org/node/1886045849).